File: 0680 Valid Palindrome II.java
class Solution {
    public boolean validPalindrome(String s) {
        char[] arr = s.toCharArray();
        for (int i = 0, j = arr.length - 1; i < j; ++i, --j)
            if (arr[i] != arr[j]) return isp(arr, i + 1, j) || isp(arr, i, j - 1);
        return true;
    }

    private boolean isp(char[] arr, int l, int r) {
        for (; l < r; ++l, --r)
            if (arr[l] != arr[r]) return false;
        return true;
    }
}

File: 0680 Valid Palindrome II.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        def isp(s: str, l: int, r: int) -> bool:
            while l < r:
                if s[l] != s[r]: return False
                l, r = l + 1, r - 1
            return True
                
        i, j = 0, len(s) - 1
        while i < j:
            if s[i] != s[j]: return isp(s, i + 1, j) or isp(s, i, j - 1)
            i, j = i + 1, j - 1
        return True